### PR TITLE
maps locator strategy used in java-client to the locator strategy in …

### PR DESCRIPTION
**Pull request by rafael-chavez**
…the WebDriverAgent

Changed signature to match that of super method. Also this matches the existing appium-ios-driver execute command [method sig](https://github.com/appium/appium-ios-driver/blob/master/lib/driver.js#L188) .

### CODE REVIEW

**Comment by rafael-chavez** [comment]
@Jonahss @jlipps there is a code signing error on your travis ci:

```
>> Code Sign error: No code signing identities found: No valid signing identities (i.e. certificate and private key pair) were found.
```

**Comment by rafael-chavez** [comment]
@jlipps @Jonahss any update on the travis ci environment?

**Comment by jlipps** [comment]
@Jonahss @imurchie do you guys have any idea about this? maybe it expired?

File: lib/driver.js [hunk 1]
```diff
@@ -240,13 +240,20 @@ class WebDriverAgentDriver extends BaseDriver {
...
     if (this.shouldProxyToSafari(cmd)) {
       let safariDriver = await this.getSafariDriver();
-      return await safariDriver.executeCommand(...arguments);
+      return await safariDriver.executeCommand(...args);
     }
+    //check to see if we are searching via predicate string
+    var locatorStrategy = args[1].req.body.using;
 
-    return await super.executeCommand(...arguments);
+    if ("-ios predicate string" === locatorStrategy) {
```
Comment (jlipps) [inline lib/driver.js]: we typically do equality checks the other way around (variable first), mind switching?

File: lib/driver.js [hunk 2]
```diff
@@ -240,13 +240,20 @@ class WebDriverAgentDriver extends BaseDriver {
            !_.includes(commandsToNotProxyToSafari, command);
   }
 
-  async executeCommand (cmd) {
+  async executeCommand (cmd, ...args) {
     if (this.shouldProxyToSafari(cmd)) {
       let safariDriver = await this.getSafariDriver();
-      return await safariDriver.executeCommand(...arguments);
+      return await safariDriver.executeCommand(...args);
```
Comment (imurchie) [inline lib/driver.js]: This ignores the command in `cmd`. Do we want to send `(cmd, ...args)`?

File: lib/driver.js [hunk 3]
```diff
@@ -240,13 +240,20 @@ class WebDriverAgentDriver extends BaseDriver {
...
     }
+    //check to see if we are searching via predicate string
+    var locatorStrategy = args[1].req.body.using;
```
Comment (imurchie) [inline lib/driver.js]: Use `let`.

File: lib/driver.js (hunk 2 above)
Comment (rafael-chavez) [inline lib/driver.js]: @imurchie previous implementation didnt pass in cmd. but to be consistent this should pass in cmd. Ill update.

File: lib/driver.js (hunk 3 above)
Comment (rafael-chavez) [inline lib/driver.js]: will do.

File: lib/driver.js (hunk 1 above)
Comment (rafael-chavez) [inline lib/driver.js]: will do.

**Comment by rafael-chavez** [comment]
@Jonahss @imurchie Travis CI failing with this error now:

```
-mand 'xcrun --sdk iphonesimulator --show-sdk-version' timed out after 3000ms
```

**Comment by rafael-chavez** [comment]
@jlipps ^^

File: lib/driver.js (hunk 2 above)
Comment (imurchie) [inline lib/driver.js]: The previous solution use `arguments`, which is _all_ the arguments. This new `args` is everything except `cmd`, which is separated out.

**Comment by imurchie** [comment]
I'm looking into this, but Travis is not starting jobs, so it's taking a while.

**Comment by rafael-chavez** [comment]
Thanks @imurchie

**Comment by rafael-chavez** [comment]
Can we fix this?

**Comment by jlipps** [comment]
@imurchie if we can't fix i'm fine merging as i think the code itself is OK

**Comment by imurchie** [comment]
I can't get the tests to pass locally, myself. The errors in Travis mostly are the test apps not being able to be built. It's on my plate to re-re-write the install script without needing babel. It causes more headaches than it is worth the way it is.

**Comment by rafael-chavez** [comment]
@imurchie i personally had to write a method in the java-client to extract the test apps from their zip container as they were not building with xcuitest driver. Not sure if this is the same issue here as i dont see the console output.

**Comment by rafael-chavez** [comment]
@jlipps i can we merge this? i have another pull for the scrolling mechanism we were talking about [here](https://github.com/appium/node-mobile-json-wire-protocol/pull/50)

**Comment by jlipps** [comment]
taking a look at this code now, why are we doing this stuff in executeCommand rather than overriding the element finding code itself?

**Comment by rafael-chavez** [comment]
Well that makes sense if i was making a change to the appium-ios-driver, since there are helpers and commands that we export (which essentially are commands to the driver object).

However for this driver, all commands are a proxyReqRes call, for which we pass in the appropriate req parameter. If i dont modify here, then i would have to do it in the proxyReqRes call.

**Comment by jlipps** [comment]
@rafael-chavez what I would imagine we would do is:

1) add the findElement command(s) to the commands files for this driver
2) add the findElement command(s) to the avoidProxy list
3) in the implementation of findElement, remap the locator strategy, and then simply proxy to WDA at that point, something kind of like:

```js
commands.findElement = async function (strategy, locator, /* whatever the args actually are */) {
  if (strategy == "foo") {
    strategy = "bar"; // redefine appropriately
  }
  // now we proxy this specific command to wda with the appropriate JSON body from JSONWP
  return await this.wda.jwproxy.command('/element', 'POST', {using: strategy, locator});  
}
```

**Comment by rafael-chavez** [comment]
@jlipps ok, i will need to create the find.js files (you mention to add it to them but these dont exist for this driver, thats where my confusion was).

This is what i did for the Scrolling command, so this will fit in nicely with those changes, ill close this out and open a new pull request with along with the scrolling.

**Comment by jlipps** [comment]
@rafael-chavez perfect, thanks for interpreting my terse comments and figuring out the architecture on your own. looking forward to more contributions from you :-)

**Comment by rafael-chavez** [comment]
No problem, Thanks for being patient reviewing!

**Pull request closed without merging**